libs/cx/src/lib/side-navigation/side-navigation.component.ts

```diff
@@ -0,0 +1,146 @@
+    AfterViewInit,
+    Component,
+    ContentChild,
+    HostBinding,
+    HostListener,
+    Input,
+    OnInit,
+    QueryList,
+    ViewChildren,
+    ViewEncapsulation
+} from '@angular/core';
+import { NestedListComponent } from '@fundamental-ngx/cx/nested-list';
+import { NestedListKeyboardService } from '@fundamental-ngx/cx/nested-list';
+import { SideNavigationUtilityDirective } from './side-navigation-utility.directive';
+import { SideNavigationMainComponent } from './side-navigation-main.component';
+import { SideNavigationModel } from './side-navigation-model';
+import { PreparedNestedListComponent } from '@fundamental-ngx/cx/nested-list';
+import { NestedListStateService } from '@fundamental-ngx/cx/nested-list';
+
+/**
+ * The side-navigation is a wrapping component representing
+ * a left navigation that can always display or expand/collapse using the menu icon within the global navigation.
+ */
+@Component({
+    // eslint-disable-next-line @angular-eslint/component-selector
+    selector: 'fdx-side-nav',
+    templateUrl: './side-navigation.component.html',
+    styleUrls: ['side-navigation.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    providers: [NestedListKeyboardService, NestedListStateService]
+})
+export class SideNavigationComponent implements AfterContentInit, AfterViewInit, OnInit {
+    /**
+     * Side navigation configuration, to pass whole model object, instead of creating HTML from scratch
+     */
+    @Input()
+    sideNavigationConfiguration: SideNavigationModel;
+
+    /** Whether condensed mode is included */
+    @Input()
+    @HostBinding('class.fdx-side-nav--condensed')
+    condensed = false;
+
+    /** Prevents the side navigation from truncating or wrapping, extending the width to its longest label. */
+    @Input()
+    dynamicWidth = false;
+
+    /**
+     * The screen width, in pixels, at which to automatically collapse the side navigation on window resize.
+     */
+    @Input()
+    collapseWidth: number;
+
+    /**
+     * Whether this side nav should display as narrow. Note this can only be used with side nav items that have icons.
+     */
+    @Input()
+    narrow = false;
+
+    /** Whether clicking on elements should change selected state of items */
+    @Input()
+    set selectable(selectable: boolean) {
+        this.nestedListState.selectable = selectable;
+    }
+
+    /** @hidden */
+    @ContentChild(SideNavigationUtilityDirective)
+    sideNavUtility: SideNavigationUtilityDirective;
+
+    /** @hidden */
+    @ContentChild(SideNavigationMainComponent, { descendants: true })
+    sideNavMain: SideNavigationMainComponent;
+
+    /** @hidden */
+    @ViewChildren(PreparedNestedListComponent)
+    preparedNestedList: QueryList<PreparedNestedListComponent>;
+
+    /** @hidden */
+    constructor(private keyboardService: NestedListKeyboardService, private nestedListState: NestedListStateService) {
+        this.keyboardService.refresh$.subscribe(() => {
```

Comment:
Finalize this somehow. Either with KeyboardService's ngOnDestroy complete, either unsubscribe from this class